Math/NumberTheory/Primes/Factorisation/QuadraticSieve.hs

```diff
@@ -212,19 +259,22 @@ convertToList (SomeKnown solution) = listBits $ SU.fromSized $ unDBVector soluti
 
 -- Given a solution, it computes the product of the numbers in the first
 -- component of the tuple.
-findFirstSquare :: Integer -> V.Vector Integer -> [Int] -> Integer
-findFirstSquare n values = foldr construct 1
-  where
-    construct solutionIndex previous = ((values V.! solutionIndex) * previous) `mod` n
+-- Consider using Mod for clarity.
```

Comment:
This is a good idea, let's return `Mod n` instead of `Integer` here and in `findSecondSquare`.